src/conmon.c

```diff
@@ -301,9 +305,12 @@ int main(int argc, char *argv[])
 	if (csname != NULL) {
 		g_unix_fd_add(console_socket_fd, G_IO_IN, terminal_accept_cb, csname);
 		/* Process any SIGCHLD we may have missed before the signal handler was in place.  */
-		check_child_processes(pid_to_handler);
-		if (!opt_exec || !opt_terminal || container_status < 0)
+		if (!opt_exec || !opt_terminal || container_status < 0) {
+			GHashTable *cache = g_hash_table_new_full(g_int_hash, g_int_equal, g_free, g_free);
+			data.cache = cache;
+			g_idle_add(check_child_processes_cb, &data);
```

Comment:
can't we unconditionally add this? it looks like we add it the same way before both loops, we just update the cache in this case. However, since we're dealing with the address of data, it shouldn't matter we update cache here

Reply:
unconditionally as in adding it after defining data, not here as well as line 419

Reply:
the `check_child_processes_cb` callback runs only once (it returns `G_SOURCE_REMOVE`), so we need to make sure it is registered for both loops.